File: Ugly_Number/Nth_Ugly_Number.kt
import java.util.*

// Checking whether number is Ugly Number or not
fun UglyNum(number: Int):Int {
    var x = 0
    var number = number
   while (number != 1) {
       if (number % 2 == 0) {
           number /= 2
       }
       else if (number % 3 == 0) {
           number /= 3
       }
       else if (number % 5 == 0) {
           number /= 5
       }
       else {
           x = -1
           break
       }
   }
   return x
}

// Finding Nth Ugly Number
fun NthUglyNum(num: Int):Int {
    var checkNum = 0
    var count = 0
    while (count < num ) {
        checkNum += 1
        var NUM = UglyNum(checkNum)
        if (NUM == 0) {
            count += 1
        }
        else {
            continue
        }
    }
    return checkNum
}

// Driver function for Ugly Number
fun main() {
    val scanner = Scanner(System.`in`)
    println("Please Enter an Integer :")
    val num = scanner.nextInt()
    var result = NthUglyNum(num)
    println("$num th Ugly Number is $result.")
}

// Input:- Please Enter an Integer : 30
// Output:- 30 th Ugly Number is 80.

File: Ugly_Number/Nth_Ugly_Number.rb
# Function that checks whether the Number is ugly or not
def Is_Ugly(x)
    m = 1
    while x != 1
        if x % 2 == 0
            x = x/2
        elsif x % 3 == 0
            x = x/3
        elsif x % 5 == 0
            x = x/5
        else
            m = 0
            break
        end
    end
    return m
end

print("Enter the Nth number")
# User Input of Nth number
n = gets.chomp.to_i
i = 1

# Finds the Nth ugly Number
while n != 1
    if Is_Ugly(i+1) == 1
        n = n-1
    end
    i = i+1
end

# Prints the ugly Number
print(i)

=begin
input:
Enter the Nth number: 10
output:
12
=end
